### PR TITLE
Import Envoy Mobile repo

### PR DESCRIPTION
**Pull request by jpsim**
This re-imports the Envoy Mobile repo, this time preserving git history so that `git log` and `git blame` remain useful.

* First commit deletes the previously imported `mobile` directory
* The next 1,783 commits are cherry-picked from https://github.com/envoyproxy/envoy-mobile after running `git-filter-repo --to-subdirectory-filter mobile` using https://github.com/newren/git-filter-repo
* The last commit re-applies the changes to `mobile/` that were done in https://github.com/envoyproxy/envoy in the last week

Part of https://github.com/envoyproxy/envoy/issues/23758

### CODE REVIEW

**Comment by jpsim** [comment]
Looks like previously a 12MB aar file was accidentally checked in. This removes it.

**Review by alyssawilk — DISMISSED** [review]

**Comment by RyanTheOptimist** [comment]
Thank you so much for doing this!

File: mobile/test/kotlin/apps/experimental/.bazelproject [hunk 1]
```diff
@@ -0,0 +1,23 @@
+workspace_type: android
+
+directories:
+  -bazel-bin
```
Comment (phlax) [inline mobile/test/kotlin/apps/experimental/.bazelproject:4]: ill clean up the lints after

**Review by Augustyniak — DISMISSED** [review]

**Review by phlax — DISMISSED** [review]
lgtm, thanks @jpsim

File: mobile/test/kotlin/apps/experimental/.bazelproject (hunk 1 above)
Comment (phlax) [reply mobile/test/kotlin/apps/experimental/.bazelproject:4]: ah - apologies!

**Review by phlax — APPROVED** [review]
lgtm, thanks again

**Pull request merged**